components/operator/internal/controllers/stack/delete/brokers.go

```diff
@@ -0,0 +1,70 @@
+package delete
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	"github.com/formancehq/operator/apis/stack/v1beta3"
+	"github.com/formancehq/operator/internal/controllers/stack/storage/nats"
+	"github.com/go-logr/logr"
+)
+
+var (
+	natsClientId = "12"
+)
+var (
+	ErrCast = errors.New("cannot cast interface to string")
+)
+
+func DeleteByBrokers(c *v1beta3.Configuration, stackName string, subjectService []string, logger logr.Logger) error {
```

Comment:
No need to use reflection as you know the structure of the configuration

Reply:
Same comment again :D
You can simply use configuration.Spec.Broker.Nats != nil (something like this)